src/block-data-providers/esplora.provider.ts

```diff
@@ -162,33 +162,59 @@ export class EsploraProvider
     }
 
     private async processBlock(height: number, hash: string) {
+        const state = await this.getState();
+        const lastProcessedTxIndex = state?.lastProcessedTxIndex ?? 0;
+
         const txids = await this.getTxidsForBlock(hash);
 
-        for (let i = 1; i < txids.length; i += this.batchSize) {
+        for (
+            let i = lastProcessedTxIndex;
```

Comment:
I think this should be `lastProcessedTxIndex + 1`. If the last processed index was `10`, we want to start from `11`. The default vault should be `-1`